daemon/irc_announce.c

```diff
@@ -131,11 +163,80 @@ static void handle_irc_privmsg(struct ircstate *istate, const struct privmsg *ms
 		       atoi(splits[7]), atoi(splits[8]), 6);
 }
 
+static void handle_node_announcement(
+	struct ircstate *istate,
+	const struct privmsg *msg,
+	char **splits)
+{
+	struct pubkey *pk = talz(msg, struct pubkey);
+	char *hostname = tal_strdup(msg, splits[2]);
+	int port = atoi(splits[3]);
+
+	if (!pubkey_from_hexstr(istate->dstate->secpctx, splits[1], strlen(splits[1]), pk) || port < 1)
+		return;
+
+	if (!verify_signed_privmsg(istate, pk, msg)) {
+		log_debug(istate->log, "Ignoring node announcement from %s, signature check failed.",
+			  splits[1]);
+		return;
+	}
+
+	add_node(istate->dstate, pk, hostname, port);
```

Comment:
I think it's better if add_node doesn't steal, but copies hostname.  Taking ownership of a pointer is a big deal, and so I prefer not to do it subtly.